pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -385,9 +385,13 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 		} else {
 			// Dual Stack
 			if features.EnableDualStack {
+				c.DnsLookupFamily = cluster.Cluster_V4_PREFERRED
 				// If dual-stack, it may be [IPv4, IPv6] or [IPv6, IPv4]
 				// using Cluster_ALL to enable Happy Eyeballsfor upstream connections
-				c.DnsLookupFamily = cluster.Cluster_ALL
+				ipFamily := networkutil.CheckIPFamilyTypeForFirstIPs(cb.proxyIPAddresses)
```

Comment:
Why check the first ip?For dual stack, can we always set `Cluster_ALL` ?

Reply:
Hi @hzxuzhonghu, it seems that `Cluster_ALL` can not be always the correct one. It may depend on environment, I can make it okay in my environment, but it's not working for some other environments.